packages/upload-api/src/ucan/conclude.js

```diff
@@ -0,0 +1,163 @@
+import * as API from '../types.js'
+import { provide } from '@ucanto/server'
+import { Receipt } from '@ucanto/core'
+import * as W3sBlob from '@web3-storage/capabilities/web3.storage/blob'
+import * as HTTP from '@web3-storage/capabilities/http'
+import { conclude } from '@web3-storage/capabilities/ucan'
+import { equals } from 'uint8arrays/equals'
+import { DecodeBlockOperationFailed } from '../errors.js'
+
+/**
+ * @param {API.ConcludeServiceContext} context
+ * @returns {API.ServiceMethod<API.UCANConclude, API.UCANConcludeSuccess, API.UCANConcludeFailure>}
+ */
+export const ucanConcludeProvider = ({
+  id,
+  receiptsStorage,
+  tasksStorage,
+  tasksScheduler,
+}) =>
+  provide(conclude, async ({ invocation }) => {
+    const receipt = getConcludeReceipt(invocation)
+
+    // Verify invocation exists failing with ReceiptInvocationNotFound
+    const ranInvocation = receipt.ran
+    const httpPutTaskGetRes = await tasksStorage.get(ranInvocation.link())
+    if (httpPutTaskGetRes.error) {
+      return httpPutTaskGetRes
+    }
+
+    // Store receipt
+    const receiptPutRes = await receiptsStorage.put(receipt)
+    if (receiptPutRes.error) {
+      return {
+        error: receiptPutRes.error,
+      }
+    }
+
+    // THIS IS A TEMPORARY HACK
+    // Schedule `blob/accept` if there is a `http/put` capabilities
+    // inside the invocation that this receipt comes from
+    const scheduleRes = await Promise.all(
+      httpPutTaskGetRes.ok.capabilities
+        // Go through invocation tasks and get all `http/put`
+        .filter((cap) => cap.can === HTTP.put.can)
+        // @ts-expect-error body exists in `http/put` but unknown type here
+        .map(async (/** @type {API.HTTPPut} */ cap) => {
+          // Get triggering task (blob/allocate) by checking blocking task from `url`
+          /** @type {API.UnknownLink} */
+          // @ts-expect-error ts does not know how to get this
+          const blobAllocateTaskCid = cap.nb.url['ucan/await'][1]
```

Comment:
I think this reads better

```suggestion
          const [,blobAllocateTaskCid] = cap.nb.url['ucan/await']
```